x-pack/filebeat/input/http_endpoint/validate.go

```diff
@@ -52,3 +73,33 @@ func (v *apiValidator) ValidateHeader(r *http.Request) (int, error) {
 
 	return 0, nil
 }
+
+func (v *apiValidator) ValidateHmac(r *http.Request) (int, error) {
+	if v.hmacHeader != "" && v.hmacToken == "" {
```

Comment:
Just to nit pick, since `hmacHeader` seems to be mandatory I would early return if empty, then check for `hmacToken` being empty, and then doing the rest ouside of any if, IMO the logic would look a bit more clear, but feel free to ignore the comment.

Reply:
Hmm they are both mandatory, but you are correct, the code also only checks if header is set but not token, and not the other way around. They both need to be configured because the HMAC signature needs to be included somewhere in the request. So if token != null, we still have to specify where the token is, and the other way around.

Have you ever seen usecases in which HMAC would be included somewhere else? Like a JSON field?